src/common/health.rs

```diff
@@ -199,28 +220,32 @@ impl Task {
         //
         // Total nodes: 5
         // Required: 5 / 2 = 2
-        let required_commit_indices_count = peer_address_by_id.len() / 2;
+        let sufficient_commit_indices_count = peer_address_by_id.len() / 2;
 
+        // *Wait* for `total nodex / 2` successful responses...
         let mut commit_indices: Vec<_> = (&mut requests)
-            .take(required_commit_indices_count)
+            .take(sufficient_commit_indices_count)
             .collect()
             .await;
 
+        // ...and also collect any additional responses, that we might have *already* received
         while let Ok(Some(resp)) = time::timeout(Duration::ZERO, requests.next()).await {
             commit_indices.push(resp);
         }
 
-        if commit_indices.len() >= required_commit_indices_count {
-            let cluster_commit_index = commit_indices
-                .into_iter()
-                .map(|resp| resp.into_inner().commit)
-                .max()
-                .unwrap_or(0);
-
-            Some(cluster_commit_index as _)
-        } else {
-            Some(0)
-        }
+        // Find the maximum commit index among all responses.
+        //
+        // Note, that we progress even if most (or even *all*) requests failed (e.g., because all
+        // other nodes are unavailable or they don't support `GetConsensusCommit` gRPC API).
+        //
+        // So this check is not 100% reliable and can give a false-positive result!
+        let cluster_commit_index = commit_indices
+            .into_iter()
+            .map(|resp| resp.into_inner().commit)
+            .max()
+            .unwrap_or(0);
+
+        Some(cluster_commit_index as _)
```

Comment:
Also, some explanation for the changes.

_For some reason_, we decided to return 0 (which means "ignore cluster commit check"), if we did not get *at least* `required_commit_indices_count` successful responses... which is silly?

Unless someone somehow added a node from *another* cluster to the existing one, or manually modified node data on disk, or *manually* break the cluster in some elaborate way, getting *any* consensus commit is better than getting none.

So I removed this nonsense logic, and I always return `max` of *any* commit indices we have. We only return 0 if *all` requests failed.

Reply:
I may speculate that we did it for the compatibility with versions that didn't support sharing of the consensus version. Anyway, I don't see the reason for keeping the old logic either.